Modules/Challenges/test/Challenges.API.Tests.Integration/StepDefinitions/ChallengesApiStepDefinitions.cs

```diff
@@ -65,7 +68,9 @@ public async Task GivenAChallengeC()
     public async Task WhenAPOSTRequestIsSentToTheChallengesEndpointWith(Table table)
     {
         var requestConfiguration = table.CreateInstance<RequestConfiguration>();
-        _challengeResponse = await _challengeApi.CreateChallenge(requestConfiguration);
+        var mergedRequestConfiguration = _mapper.Map(_requestConfiguration, requestConfiguration);
```

Comment:
I appreciate your effort to do some research on this topic. And in most cases this would probably be the best solution.
But in this case, the merge is pretty simple and contains only a few properties. Introducing AutoMapper for this sole purpose is a bit overkill. That's why I would have used a simple method on the RequestConfiguration:

```c#
public void SupplementWith(RequestConfiguration other) 
{
    this.Authenticate ?= other.Authenticate; 
    this.AcceptHeader ?= other.AcceptHeader;
    this.ContentTypeHeader ?= other.ContentTypeHeader;
    this.Content ?= other.Content;
}
``` 
You need to make all the properties nullable to be able to do this.

And to call it:

```c#
requestConfiguration.SupplementWith(_requestConfiguration);
```